src/Data/Record/Ordered.idr

```diff
@@ -13,13 +14,19 @@ ArgList : Type
 ArgList = FreshList String String.(#)
 
 public export
-Fields : Type -> Type
-Fields a = FreshList (String, a) ((#) `on` fst)
+Field : (String -> Type) -> Type
+Field a = (nm : String ** a nm)
```

Comment:
Should `nm` be runtime irrelevant? It's useful to have it, sometimes, but must we have it always?

Since we can ask this question, we ought to.

Reply:
I think we do want `name` to be around at runtime when the fields are.
We use it to print the usage information for subcommands and flags for instance .

Reply:
It depends how we manipulate `Fields`, esp. if we're going to apply `map`s to them a lot, we might not want to actually carry the `nm` around for all the derived structure.

(If we need `nm` at runtime, we can store a string + a proof it's equal to `nm` instead for the `Fields` we need it for.)

Reply:
(This might be a premature optimisation, too. Maybe all we need to do is leave a TODO note to our future-selves and ask it again after we used the library more.)